core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -131,6 +132,25 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
     set_auto_chunk_size();
   }
 
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  template <typename IndexType,
+            std::enable_if_t<(std::is_integral_v<IndexType> &&
+                              !std::is_same_v<IndexType, member_type> &&
+                              std::is_convertible_v<IndexType, member_type>),
+                             bool> = false>
```

Comment:
Why did you chose to add a new templated constructors instead of updating the existing ones?

Reply:
There wasn't a strong reason for it. Part of the minor reasoning was for better readability so that the existing constructor isn't going to be heavily interleaved by multiple `ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4`.
But I just realized that the constructor with an execution space should also have this check. I'll make the updates and integrate this conversion into the existing constructors instead.

Reply:
Moved this change into existing constructors. [67fcdb0](https://github.com/kokkos/kokkos/pull/6754/commits/67fcdb02ef547355bce81dea0fc8942842c13680).